spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-sample/src/test/java/com/example/PubSubApplicationTests.java

```diff
@@ -0,0 +1,276 @@
+						+ "to enable them. ",
+				System.getProperty("it.pubsub"), is("true"));
+
+		projectName = ProjectName.of(ServiceOptions.getDefaultProjectId()).getProject();
+		topicAdminClient = TopicAdminClient.create();
+		subscriptionAdminClient = SubscriptionAdminClient.create();
+	}
+
+	@AfterClass
+	public static void cleanupPubsubClients() {
+		if (topicAdminClient != null) {
+			topicAdminClient.close();
+		}
+
+		if (subscriptionAdminClient != null) {
+			subscriptionAdminClient.close();
+		}
+	}
+
+	@Before
+	public void initializeAppUrl() throws IOException {
+		this.appUrl = "http://localhost:" + this.port;
+	}
+
+	@Before
+	@After
+	public void cleanupPubsubTestResources() {
+		List<String> projectTopics = getTopicNamesFromProject();
+		String testTopicName = ProjectTopicName.format(projectName, SAMPLE_TEST_TOPIC);
+		if (projectTopics.contains(testTopicName)) {
+			topicAdminClient.deleteTopic(testTopicName);
+		}
+
+		List<String> testSubscriptions = ImmutableList.of(SAMPLE_TEST_SUBSCRIPTION1, SAMPLE_TEST_SUBSCRIPTION2);
+		for (String testSubscription : testSubscriptions) {
+			String testSubscriptionName = ProjectSubscriptionName.format(
+					projectName, testSubscription);
+			List<String> projectSubscriptions = getSubscriptionNamesFromProject();
+			if (projectSubscriptions.contains(testSubscriptionName)) {
+				subscriptionAdminClient.deleteSubscription(testSubscriptionName);
+			}
+		}
+	}
+
+	@Test
+	public void testCreateAndDeleteTopicAndSubscriptions() {
+		String projectTopicName = ProjectTopicName.format(projectName, SAMPLE_TEST_TOPIC);
+		String projectSubscriptionName = ProjectSubscriptionName.format(projectName, SAMPLE_TEST_SUBSCRIPTION1);
+
+		createTopic(SAMPLE_TEST_TOPIC);
+		await().atMost(PUBSUB_CLIENT_TIMEOUT_SECONDS, TimeUnit.SECONDS).untilAsserted(
+				() -> {
+					List<String> projectTopics = getTopicNamesFromProject();
+					assertThat(projectTopics).contains(projectTopicName);
+				});
+
+		createSubscription(SAMPLE_TEST_SUBSCRIPTION1, SAMPLE_TEST_TOPIC);
+		await().atMost(PUBSUB_CLIENT_TIMEOUT_SECONDS, TimeUnit.SECONDS).untilAsserted(
+				() -> {
+					List<String> subscriptions = getSubscriptionNamesFromProject();
+					assertThat(subscriptions).contains(projectSubscriptionName);
+				});
+
+		deleteSubscription(SAMPLE_TEST_SUBSCRIPTION1);
+		await().atMost(PUBSUB_CLIENT_TIMEOUT_SECONDS, TimeUnit.SECONDS).untilAsserted(
+				() -> {
+					List<String> subscriptions = getSubscriptionNamesFromProject();
+					assertThat(subscriptions).doesNotContain(projectSubscriptionName);
+				});
+
+		deleteTopic(SAMPLE_TEST_TOPIC);
+		await().atMost(PUBSUB_CLIENT_TIMEOUT_SECONDS, TimeUnit.SECONDS).untilAsserted(
+				() -> {
+					List<String> projectTopics = getTopicNamesFromProject();
+					assertThat(projectTopics).doesNotContain(projectTopicName);
+				});
+	}
+
+	@Test
+	public void testReceiveMessage() {
+		createTopic(SAMPLE_TEST_TOPIC);
+		createSubscription(SAMPLE_TEST_SUBSCRIPTION1, SAMPLE_TEST_TOPIC);
```

Comment:
Hmm, so I haven't encountered this problem but I think it should be fine; it looks like the createTopic and createSubscription calls made by the clients are blocking and so each rest endpoint only returns after the call succeeds/fails.

I added await to the previous create/delete steps because it looked like there was some latency between creating a topic and then having it appear in a subsequent .ListTopics() from the pubsub client.

here is the latest run: https://travis-ci.org/spring-cloud/spring-cloud-gcp/builds/443736825